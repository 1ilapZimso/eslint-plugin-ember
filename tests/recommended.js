'use strict';

const rules = require('../lib').rules;

describe('recommended rules', () => {
  it('has the right list', () => {
    const errors = [];

    for (const name of Object.keys(rules)) {
      // eslint-disable-next-line jest/no-if
      if (rules[name].meta.docs.recommended) {
        errors.push(name);
      }
    }

    expect(errors).toMatchSnapshot();
  });
});
