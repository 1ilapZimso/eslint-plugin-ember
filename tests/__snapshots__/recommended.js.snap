// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recommended rules has the right list 1`] = `
Array [
  "avoid-leaking-state-in-ember-objects",
  "avoid-using-needs-in-controllers",
  "closure-actions",
  "jquery-ember-run",
  "new-module-imports",
  "no-arrow-function-computed-properties",
  "no-assignment-of-untracked-properties-used-in-tracking-contexts",
  "no-attrs-in-components",
  "no-attrs-snapshot",
  "no-capital-letters-in-routes",
  "no-controller-access-in-routes",
  "no-deeply-nested-dependent-keys-with-each",
  "no-duplicate-dependent-keys",
  "no-ember-super-in-es-classes",
  "no-ember-testing-in-module-scope",
  "no-function-prototype-extensions",
  "no-get-with-default",
  "no-get",
  "no-global-jquery",
  "no-incorrect-calls-with-inline-anonymous-functions",
  "no-incorrect-computed-macros",
  "no-invalid-debug-function-arguments",
  "no-invalid-dependent-keys",
  "no-invalid-test-waiters",
  "no-jquery",
  "no-legacy-test-waiters",
  "no-mixins",
  "no-new-mixins",
  "no-noop-setup-on-error-in-before",
  "no-observers",
  "no-old-shims",
  "no-on-calls-in-components",
  "no-pause-test",
  "no-private-routing-service",
  "no-restricted-resolver-tests",
  "no-side-effects",
  "no-test-and-then",
  "no-test-import-export",
  "no-test-module-for",
  "no-test-this-render",
  "no-unnecessary-route-path-option",
  "no-volatile-computed-properties",
  "prefer-ember-test-helpers",
  "require-computed-macros",
  "require-computed-property-dependencies",
  "require-return-from-computed",
  "require-super-in-init",
  "routes-segments-snake-case",
  "use-brace-expansion",
  "use-ember-data-rfc-395-imports",
]
`;
